headers/types/common/common.h

```diff
@@ -4,6 +4,7 @@
 #define HEADERS_TYPES_COMMON_H_
 
 #include "enums.h"
+#include "../dungeon_mode/enums.h"
```

Comment:
Yup. Right now it's only needed for `dungeon_init::dungeon_objective`, but it wouldn't make much sense to move the `dungeon_objective` enum somewhere else since it's related to dungeons.